framework/caching/Cache.php

```diff
@@ -468,6 +505,7 @@ protected function setValues($data, $duration)
             if ($this->setValue($key, $value, $duration) === false) {
                 $failedKeys[] = $key;
             }
+            $this->endProfile($key, __METHOD__);
```

Comment:
Seems there are no tests that run with enableProfiling = true. These should be added as they are expected to fail at this line.